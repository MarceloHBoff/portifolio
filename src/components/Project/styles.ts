import styled, { css } from 'styled-components'

type ContainerProps = {
  vertical: boolean
}

export const Container = styled.div<ContainerProps>`
  ${({ theme, vertical }) => css`
    width: 100%;
    padding: ${theme.spacings.normal};

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    cursor: pointer;

    img {
      width: 100%;
      height: 700px;
      border-radius: 10px;

      ${vertical &&
      css`
        width: 35%;
        border-radius: 0px;
        margin: 0 auto;
      `}
    }
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    width: 100%;
    height: 150px;
    margin-top: -150px;
    background: ${theme.colors.secundary};
    background: rgba(0, 0, 0, 0.7);
    padding: ${theme.spacings.normal};
    border-radius: 0 0 10px 10px;
    box-shadow: 0px 0px 15px 0px rgba(0, 0, 0, 0.3);
  `}
`

export const Title = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.normal};
    font-size: ${theme.font.sizes.xxlarge};
    font-weight: ${theme.font.bold};
  `}
`

export const Description = styled.div`
  ${({ theme }) => css`
    line-height: ${theme.font.sizes.xxlarge};
    font-size: ${theme.font.sizes.medium};
  `}
`
