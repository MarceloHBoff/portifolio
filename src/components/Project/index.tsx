import Link from 'next/link'

import { Container, Content, Title, Description } from './styles'

export type ProjectType = {
  title: string
  description: string
  liveurl: string
  giturl: string
  color: string
  uid: string
  banner: {
    vertical: boolean
    url: string
    alt: string
  }
  images: {
    vertical: boolean
    url: string
    alt: string
  }[]
}

type ProjectProps = {
  project: ProjectType
}

export default function Project({ project }: ProjectProps) {
  return (
    <Link href={`/project/${project.uid}`}>
      <Container vertical={project.banner.vertical}>
        <img src={project.banner.url} alt={project.banner.alt} />

        <Content>
          <Title>{project.title}</Title>
          <Description>{project.description}</Description>
        </Content>
      </Container>
    </Link>
  )
}
