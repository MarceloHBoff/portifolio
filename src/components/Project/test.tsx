import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'

import Project from '.'

const project = {
  title: 'title',
  description: 'description',
  liveurl: '',
  giturl: '',
  color: '',
  uid: 'uid',
  banner: {
    vertical: true,
    url: '',
    alt: 'banner'
  },
  images: [
    {
      vertical: false,
      url: '',
      alt: 'alt1'
    },
    {
      vertical: true,
      url: '',
      alt: 'alt2'
    }
  ]
}

describe('<Project />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Project project={project} />)

    expect(screen.getByText(project.title)).toBeInTheDocument()
    expect(screen.getByText(project.description)).toBeInTheDocument()
    expect(screen.getByAltText(project.banner.alt)).toBeInTheDocument()

    expect(container.parentElement).toMatchSnapshot()
  })
})
