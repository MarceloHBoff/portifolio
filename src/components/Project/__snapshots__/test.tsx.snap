// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project /> should render correctly 1`] = `
.c0 {
  width: 100%;
  padding: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0 img {
  width: 100%;
  height: 700px;
  border-radius: 10px;
  width: 35%;
  border-radius: 0px;
  margin: 0 auto;
}

.c1 {
  width: 100%;
  height: 150px;
  margin-top: -150px;
  background: #222;
  background: rgba(0,0,0,0.7);
  padding: 1.5rem;
  border-radius: 0 0 10px 10px;
  box-shadow: 0px 0px 15px 0px rgba(0,0,0,0.3);
}

.c2 {
  margin-bottom: 1.5rem;
  font-size: 2.4rem;
  font-weight: 700;
}

.c3 {
  line-height: 2.4rem;
  font-size: 1.6rem;
}

<body>
  <div>
    <div
      class="c0"
    >
      <img
        alt="banner"
        src=""
      />
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          title
        </div>
        <div
          class="c3"
        >
          description
        </div>
      </div>
    </div>
  </div>
</body>
`;
